### ISSUE TITLE
Command & Conquer 3: Kane's Wrath (24810)

### ISSUE THREAD

**Issue by hr87** [ValveSoftware/Proton#477]
Works perfectly for me, played for about 1h.

Ubuntu 18.04
AMD Graphics
Default Proton settings

**Comment by kisak-valve** [comment]
Hello @hr87, please copy your system information from steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by hr87** [comment]
https://gist.github.com/hr87/1d114e64635feeb464970e6b4d7e65af

**Comment by Tibsun75** [comment]
The Game works but:
The Games Crash by pressing ALT+TAB

my gist:
https://gist.github.com/Tibsun75/516d6bb06538ef36c6dfbf98c595bc72

**Comment by kisak-valve** [comment]
Hello @Tibsun75, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box.

**Comment by Tibsun75** [comment]
... the generated logfile has 64MB. Are you sure ?

**Comment by Tibsun75** [comment]
i deleted the first logfile. But the second logfile has now 75MB...
Therefore i post only the last 100 lines:
tail -100 steam-24810.log
3370.459:0024:0025:trace:module:MODULE_InitDLL (0x7bc20000,PROCESS_DETACH,0x1) - RETURN 1
3370.464:000c:003a:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_ATTACH,(nil)) - CALL
3370.464:000c:003a:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_ATTACH,(nil)) - RETURN 1
3370.464:000c:003b:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_ATTACH,(nil)) - CALL
3370.464:000c:003b:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_ATTACH,(nil)) - RETURN 1
3370.464:000c:003c:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_ATTACH,(nil)) - CALL
3370.464:000c:003c:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_ATTACH,(nil)) - RETURN 1
3370.464:000c:003d:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_ATTACH,(nil)) - CALL
3370.464:000c:003d:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_ATTACH,(nil)) - RETURN 1
3370.471:000c:003e:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_ATTACH,(nil)) - CALL
3370.471:000c:003e:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_ATTACH,(nil)) - RETURN 1
3370.506:000f:0015:trace:module:LdrShutdownThread ()
3370.506:000f:0015:trace:module:MODULE_InitDLL (0x7fbb5dd40000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.506:000f:0015:trace:module:MODULE_InitDLL (0x7fbb5dd40000,THREAD_DETACH,(nil)) - RETURN 1
3370.507:000c:003f:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_ATTACH,(nil)) - CALL
3370.507:000c:003f:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_ATTACH,(nil)) - RETURN 1
3370.507:001d:0021:trace:module:LdrShutdownThread ()
3370.507:001d:0021:trace:module:MODULE_InitDLL (0x7f2f20c00000 L"imm32.dll",THREAD_DETACH,(nil)) - CALL
3370.507:001d:0021:trace:module:MODULE_InitDLL (0x7f2f20c00000,THREAD_DETACH,(nil)) - RETURN 1
3370.507:001d:0021:trace:module:MODULE_InitDLL (0x7f2f21a30000 L"user32.dll",THREAD_DETACH,(nil)) - CALL
3370.507:0018:001c:trace:module:LdrShutdownThread ()
3370.507:0018:001c:trace:module:MODULE_InitDLL (0x7f7839d70000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.507:0018:001c:trace:module:MODULE_InitDLL (0x7f7839d70000,THREAD_DETACH,(nil)) - RETURN 1
3370.507:001d:0021:trace:module:MODULE_InitDLL (0x7f2f21a30000,THREAD_DETACH,(nil)) - RETURN 1
3370.507:001d:0021:trace:module:MODULE_InitDLL (0x7f2f21f30000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.508:001d:0021:trace:module:MODULE_InitDLL (0x7f2f21f30000,THREAD_DETACH,(nil)) - RETURN 1
3370.573:001d:001e:trace:module:LdrShutdownProcess ()
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21c80000 L"setupapi.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21c80000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f20c00000 L"imm32.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f20c00000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21a30000 L"user32.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21a30000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:0018:0019:trace:module:LdrShutdownProcess ()
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21880000 L"version.dll",PROCESS_DETACH,0x1) - CALL
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7f7839d70000 L"rpcrt4.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21880000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f218b0000 L"gdi32.dll",PROCESS_DETACH,0x1) - CALL
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7f7839d70000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f218b0000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7f7839e10000 L"advapi32.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21f30000 L"rpcrt4.dll",PROCESS_DETACH,0x1) - CALL
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7f7839e10000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21f30000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7b420000 L"KERNEL32.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21fd0000 L"ntoskrnl.exe",PROCESS_DETACH,0x1) - CALL
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7b420000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f21fd0000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7bc20000 L"ntdll.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f22030000 L"advapi32.dll",PROCESS_DETACH,0x1) - CALL
3370.573:0018:0019:trace:module:MODULE_InitDLL (0x7bc20000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7f2f22030000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7b420000 L"KERNEL32.dll",PROCESS_DETACH,0x1) - CALL
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7b420000,PROCESS_DETACH,0x1) - RETURN 1
3370.573:001d:001e:trace:module:MODULE_InitDLL (0x7bc20000 L"ntdll.dll",PROCESS_DETACH,0x1) - CALL
3370.574:001d:001e:trace:module:MODULE_InitDLL (0x7bc20000,PROCESS_DETACH,0x1) - RETURN 1
3370.574:000f:0010:trace:module:LdrShutdownProcess ()
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7fbb5dd40000 L"rpcrt4.dll",PROCESS_DETACH,0x1) - CALL
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7fbb5dd40000,PROCESS_DETACH,0x1) - RETURN 1
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7fbb5dde0000 L"ntoskrnl.exe",PROCESS_DETACH,0x1) - CALL
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7fbb5dde0000,PROCESS_DETACH,0x1) - RETURN 1
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7fbb5de40000 L"advapi32.dll",PROCESS_DETACH,0x1) - CALL
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7fbb5de40000,PROCESS_DETACH,0x1) - RETURN 1
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7b420000 L"KERNEL32.dll",PROCESS_DETACH,0x1) - CALL
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7b420000,PROCESS_DETACH,0x1) - RETURN 1
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7bc20000 L"ntdll.dll",PROCESS_DETACH,0x1) - CALL
3370.574:000f:0010:trace:module:MODULE_InitDLL (0x7bc20000,PROCESS_DETACH,0x1) - RETURN 1
3370.576:000c:001a:trace:module:LdrShutdownThread ()
3370.576:000c:001a:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.576:000c:001a:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_DETACH,(nil)) - RETURN 1
3370.577:000c:003e:trace:module:LdrShutdownThread ()
3370.577:000c:003e:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.577:000c:003e:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_DETACH,(nil)) - RETURN 1
3370.578:000c:001f:trace:module:LdrShutdownThread ()
3370.578:000c:0022:trace:module:LdrShutdownThread ()
3370.578:000c:0016:trace:module:LdrShutdownThread ()
3370.578:000c:0011:trace:module:LdrShutdownThread ()
3370.578:000c:001f:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.578:000c:001f:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_DETACH,(nil)) - RETURN 1
3370.578:000c:0022:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.578:000c:0022:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_DETACH,(nil)) - RETURN 1
3370.578:000c:0016:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.578:000c:0016:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_DETACH,(nil)) - RETURN 1
3370.579:000c:0011:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.579:000c:0011:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_DETACH,(nil)) - RETURN 1
3370.580:000c:000e:trace:module:LdrShutdownThread ()
3370.580:000c:000e:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",THREAD_DETACH,(nil)) - CALL
3370.580:000c:000e:trace:module:MODULE_InitDLL (0x7f1606260000,THREAD_DETACH,(nil)) - RETURN 1
3370.581:000c:000d:trace:module:LdrShutdownProcess ()
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7f16061b0000 L"userenv.dll",PROCESS_DETACH,0x1) - CALL
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7f16061b0000,PROCESS_DETACH,0x1) - RETURN 1
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7f1606260000 L"rpcrt4.dll",PROCESS_DETACH,0x1) - CALL
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7f1606260000,PROCESS_DETACH,0x1) - RETURN 1
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7f16061d0000 L"advapi32.dll",PROCESS_DETACH,0x1) - CALL
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7f16061d0000,PROCESS_DETACH,0x1) - RETURN 1
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7b420000 L"KERNEL32.dll",PROCESS_DETACH,0x1) - CALL
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7b420000,PROCESS_DETACH,0x1) - RETURN 1
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7bc20000 L"ntdll.dll",PROCESS_DETACH,0x1) - CALL
3370.581:000c:000d:trace:module:MODULE_InitDLL (0x7bc20000,PROCESS_DETACH,0x1) - RETURN 1
pid 6149 != 6148, skipping destruction (fork without exec?)

**Comment by kisak-valve** [comment]
That's 100 worthless lines of the game closing. The log should compress well in an archive and you can drag and drop an archive into the comment box.

**Comment by Tibsun75** [comment]
Good idea.
Here ist the compressed file:
[steam-24810.zip](https://github.com/ValveSoftware/Proton/files/2719039/steam-24810.zip)

**Comment by jkcdarunday** [comment]
My current workaround for this is to configure wine to emulate a virtual desktop with the same resolution as your screen and disable window decorations for it. From what I understand, this puts the window switching in the emulated desktop instead of directly on the game window.

**Comment by malefice** [comment]
I have not tried Kane's Wrath expansion, but the issue does exist in the base game C&C3 Tiberium Wars, though I am not sure if I should create a new issue. The game crashes whenever you alt-tab to another running application if you are running the game in fullscreen mode. The game will not crash if you are running it in windowed mode.

Here is my system [details](https://gist.github.com/malefice/823f3986903b9febe89a4b2cf3e26c0f), and here is a much more manageable [logfile](https://github.com/ValveSoftware/Proton/files/4464904/steam-24790.log). To generate the logfile, I first cleared whatever logfile is present for the game and then started the game in fullscreen mode. Once I reached the main menu, I alt-tab to another window.

**Comment by kisak-valve** [comment]
Hello @malefice, please add your feedback about Tiberium Wars to #1187.

**Comment by kisak-valve** [comment]
How to Build Proton Without Faudio - Revert to older Wine XAudio2 implementation

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6142.
@Tribalwarfare posted on 2022-09-06T13:24:07:


Hi,

I am currently using the latest version of Proton 7.0-4 stable to play Command and Conquer 3: Kanes Wrath from the Steam Library.
The APP ID is 24810.

In single player/Skirmish it runs great, however with any version of Proton or Wine that includes Faudio the game desyncs when playing LAN or Online multiplayer games.

Steam Proton 3.7-8 was the last version that did not cause and or result in desyncs as it did not include Faudio.

Is there a method to build the current stable Steam Proton version without the Faudio module and use the older Wine XAudio2 implementation? Can someone possibly provide a patch or guide on how to disable Faudio and use the older Wine XAudio2 implementation?

Thank you for your kind assistance,
AG

**Comment by kisak-valve** [comment]
Hello @Tribalwarfare, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression, and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

**Comment by Tribalwarfare** [comment]
> Hello @Tribalwarfare, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression, and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

Hi Kisak, 

Please find attached the log as requested. Just performed a local test with two networked systems and replicated the desync error.

Kind Regards,
AG
[steam-24810.zip](https://github.com/ValveSoftware/Proton/files/9498149/steam-24810.zip)

**Comment by kisak-valve** [comment]
Thanks, no references to xaudio or faudio in the log. There very well could be a regression which has been hiding for a long time, but FAudio is most likely uninvolved.

**Comment by Tribalwarfare** [comment]
Thank you for looking into it :)

**UPDATE:** I was able to get the game running online without desyncs by porting older wine math and system libraries from proton_tkg_5.14.r0.gb71d588d.release and installing Microsoft Visual c++ 2005 to 2015 versions.

Alt-Tab fix: Rune WineCFG and enable virtual desktop.

Thank you to Valve Proton team and Tk-Glitch for making the builds available :)

I will post a download link + setup guide once I can find a host.

**Comment by AJuujarvi** [comment]
@Tribalwarfare I was looking into this just now as we would want to fix this regression but I am a bit stumped. If I try to play Online, the game says it is not connecting to any online servers. If I try to do Network, both of my devices are not able to find each other for Network LAN play. I am not familiar with this game, so there is likely something I am missing. Can you let me know what steps I need to perform this desync issue so I can further investigate it please? Is there any Port Forwarding I have to do for example?

**Comment by Tribalwarfare** [comment]
@AJuujarvi :

Hi there: The game has had no official Online play since GameSpy was shutdown. We make use of the community hosted Revora Project hosted at https://cnc-online.net/en/.

Once you have registered an account and have it activated you can download the Linux compatible installer client from their Discord as the Windows loader tool is not yet Proton compatible.

Here is my ProtonDB guide I recently wrote and also my full guide on the cnc-online discord to get things going:

https://www.protondb.com/app/24810
https://discord.com/channels/264522892455182346/528509243507081217/1018025425655570432

**TLDR:**
I have tracked the issue down to changes in the builtin wine vcruntime files provided by wine/proton.
Deleting your old 24810 compat folder, setting Proton 7.0-4 as the runner and installing the following vcruntime files fixes the issue:
vcrun2005 vcrun2008 vcrun2010 vcrun2012 vcrun2013 vcrun2019

**My Quick Guide from ProtonDB:**
"Multiplayer Desync Fix for newer Proton Versions: (works cross platform)

Delete old 24810 pfx folder /steamapps/compatdata/. Set Proton to version 7.0-4 In steam and Launch game to create a new fresh 24810 folder. Exit game.

run protontricks --no-bwrap 24810 --gui

(--no-bwrap used incase winecfg does not load)

First enable virtual Desktop Mode and set desired resolution.

Next Select Install Windows DLL or component.

Install the following vcruntimes manually approving each setup window that pop ups. vcrun2005 vcrun2008 vcrun2010 vcrun2012 vcrun2013 vcrun2019

Installer will automatically updates DLL overrides to native for the installed libraries

Game should now run Online/Lan games without desyncs.

You can get the revora online play enabler from cnc-online.net and go to their Discord to get the Linux online play installer. I wrote an entire guide there and how to.

Have fun!

Windowing:
Switching
Fixed by enabling Virtual Desktop mode via Protontricks --> Winecfg.

Instability:
Not Listed
Multiplayer Desync. Fixed by installing vcruntime components 2005 to 2019."

I would recommend reaching out to Cervanthes from the cnc-online discord channel to get the KWEnabler.exe setup files to patch in the needed Online Play components :)

His handle on the channel is @Cerv
Profile ID: Cervanthes#3136

I have sent him a DM asking him to keep an eye out for you or whether I can assist with providing the files directly.

Kind Regards,
AG

**Comment by Tribalwarfare** [comment]
@AJuujarvi : I performed a quick test for you to investigate LAN only play:

Windows players on the cnc-online forums report that LAN play works fine on local networks whether running the Revora client or not. Revora files are only requires for Online match making.

I proceeded to re-install my Manjaro Proton copy of the game without any Online Play modifications and attempted to join a LAN lobby with a Windows hosted game. I could not see any Local LAN Lobby games from the Windows Hosted session of Kanes Wrath.

This possibly implies that the Proton version of the game is not detecting/broadcasting traffic for Local LAN Match Making.

Online play does not work unless you use the patched Revora files. If you do online matchmaking works 100% fine.
Local LAN play seems to require UDP broadcast traffic of some sort.

I proceeded to Load the game up again and started a lobby on the Windows client and could see it was broadcasting network wide (255.255.255.255) + listening on UDP Port 8086.

![image](https://user-images.githubusercontent.com/110111850/189521528-6dfb1665-b25a-4138-9f0c-cb1f6be1de59.png)

It is a strong possibility WINE/Proton is not detecting/broadcasting the same own its own as my ports check on the Linux hosted game shows the port unbound:

command used: sudo ss -atulp
                                                                                                                                                                                                                     
![image](https://user-images.githubusercontent.com/110111850/189522248-680e275c-b8bd-4530-8e16-471afa648088.png)


Kind Regards,
AG

**Comment by Tribalwarfare** [comment]
@AJuujarvi: OK I have a fix for you for the LAN Play connectivity issue. Note you can re-create the de-sync issue on local LAN Play as well so you do not need to use Revora.

I was able to figure out why LAN games were not showing. It has to do with the game using the wrong Network Interface.

Wine detects all physical and virtual Interfaces and makes them available for the running game process. When accessing LAN play I noticed that it was binding to my Vware Virtual Network interface I use for my Admin tasks:
![image](https://user-images.githubusercontent.com/110111850/189523325-be563535-c3c1-45c8-9094-b7604c8d002e.png)

This is the wrong interface and was the root cause of local network and lobby game not showing.

Now what can get confusing for some people is that when you configure the network interface IP address to use Under Options-Network, it actually sets the **_GameSpyIPAddress_** IP address to use for ONLINE games and DOES not affect the interface that the game automatically binds to for LAN games.

![image](https://user-images.githubusercontent.com/110111850/189523395-062ff796-69ae-49fc-bb8c-6f5c0b0875b6.png)

As mentioned wine detects all interfaces and its an common issue on Windows installs as well for the game to bind to the wrong Address for LAN Lobby games. Due to it needing to it broadcasting UDP on the bound interface network to 255.255.255.255 if not on the correct LAN segment it will fail to see other games.

To make sure the game actually binds to the correct IP address and Interface you must add the following line to your **_compatdata/24810/pfx/drive_c/users/steamuser/AppData/Roaming/Command & Conquer 3 Kane's Wrath/Profiles/%profilenamegoeshere%/Options.ini** file.

**_IPAddress = %youractualtestnetworkIPAddress%_**

In my case: 
IPAddress = 192.168.66.222

![image](https://user-images.githubusercontent.com/110111850/189523680-3f5af3a1-2a32-49a2-91c8-2c8e60c4a3d1.png)

Starting up the game and going to lobby I could now see it was binding to the correct Interface (192.168.66.222 /enp37s) and not my Vmware Adapter. 

![image](https://user-images.githubusercontent.com/110111850/189523319-5b3cb662-67d4-4f38-9f07-f8721da9be31.png)

Games now showing in Lobby and I am able to connect:

![image](https://user-images.githubusercontent.com/110111850/189523947-3321bc21-7ea1-4c92-9792-5e96d7f5c9f8.png)


For most Windows installs Windows will use the Network Inerface preferences setting and usually correctly bind to the proper interface. Due to Linux and Wine in general, looks like we need to tell the game manually which is the correct interface to use as Linux has several default loopback and private Socket interfaces.

This goes even double so if you like myself run virtual machines and other services such as VPN interfaces etc.

So this is the fix to get your game running for LAN play so make sure to add the IPAddress = parameter to your Profiles Options.ini file and test for the desync issue :D

Kind Regards,
AG

**Comment by Tribalwarfare** [comment]
@AJuujarvi : I have been granted permission to provide you with a copy of the cnc-online Kanes Wrath Online Match making installer. It is a custom version they make available to me for Proton use. Might it be possible to include this for future testing of Online based match making and play. The developer of the Revora Online client tool have asked that you reach out to them on **_contact@cnc-modbot.com_** as they would also like to see if it would be possible to make it easier for new players to access the tools/files.

Let me know when you want access and where I can upload it securely for you to access.

Kind Regards,
AG

**Comment by AJuujarvi** [comment]
@Tribalwarfare I'm currently unable to test this as I experienced a physical injury that prevents me from being in our office for at least a week. I'll take a look at this as soon as I have the tools I need. Just wanted to let you know I saw this, and will check this out via LAN play asap. Since the original report says the regression happened with LAN+Online, my hope is that a fix for one will fix both.

In regards to making it easier for new players to access the tools/files, I would have to recommend getting the Revora Online client onto the Steam store as a Tool if possible.

**Comment by Tribalwarfare** [comment]
> @Tribalwarfare I'm currently unable to test this as I experienced a physical injury that prevents me from being in our office for at least a week. I'll take a look at this as soon as I have the tools I need. Just wanted to let you know I saw this, and will check this out via LAN play asap. Since the original report says the regression happened with LAN+Online, my hope is that a fix for one will fix both.
> 
> In regards to making it easier for new players to access the tools/files, I would have to recommend getting the Revora Online client onto the Steam store as a Tool if possible.

Hi, I hope you recover well and thank you for getting back to me. Really appreciated. I will pass the message along about the tools part to see whey they think :)

**Comment by AJuujarvi** [comment]
@Tribalwarfare Unfortunately I have been unable to reproduce the issue with LAN play. Granted I wasn't sure exactly what it was I was looking for. Is the audio desyncing from the visual effects? Are the players desyncing from each other? Is this an immediate desync or do I have to play the game for a bit before there is a desync? I played for about 20 minutes. I was using the latest Bleeding-Edge Experimental build for testing. Any additional info you have for what exactly you experience can be helpful since I am not reproducing any issue.

**Comment by Tribalwarfare** [comment]
@AJuujarvi : Hi, hope you have recovered well :)
The desync occurs when playing with two or more players and one is running Windows and the Other Linux. Desync usually occurs around the 5-8 minute mark.

Were you testing Proton to Proton?

Kind Regards,
AG

**Comment by AJuujarvi** [comment]
@Tribalwarfare Sorry for the delay! A lot of things came up.

It was between Proton to Proton. I had a chance to check Proton to Windows just now and I also did not have any desync issues. I tested with the latest Proton Experimental. Specifically build "experimental-bleeding-edge-7.0-28200-20221102-pfecfc5-we924438-d892f67-v4df366".

This was tested with LAN play on 2 different networks for both a Proton to Proton connection and a Proton to Windows connection. Each one I tested for about 30 minutes each time, 3 times each. I was unable to replicate the issue any of those times.

Are you still encountering the issue with LAN play and with Revora?

**Comment by Wedge009** [comment]
Just picked this up today, seems to work just as I remember it on Windows, running on Proton 8.0-2. Only issue I had was switching windows (Alt-Tab) results in a blank screen when returning to the game.

**Comment by kisak-valve** [comment]
Command and Conquer 3: Kane's Wrath / 24810 / Black screen + loading freezes

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7436.
@st4r-w4rs posted on 2024-01-23T14:38:37:

# Compatibility Report
- Name of the game with compatibility issues: **Command and Conquer 3: Kane's Wrath**
- Steam AppID of the game: **24810**

## System Information
- GPU: NVIDIA GeForce GTX 1050 Ti
- Video driver version: 535.146.02
- Kernel version: 5.15.0-91-generic
- Link to full system information report as Gist: [50a6c222506c7add0d88b3eb620d2b38](https://gist.github.com/st4r-w4rs/50a6c222506c7add0d88b3eb620d2b38)
- Proton version: Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.


[steam-24810.log](https://github.com/ValveSoftware/Proton/files/14025425/steam-24810.log)

## Symptoms 
1) Switching ingame resolution stucks the game within a black screen, then it freezes forever.
2) When trying launching the game (either campaingn or skirmish), the loading bar goes up to 80%, then it freezes forever.

_Same symptoms on Tiberium Wars!_
These games were working out of the box one week ago on my machine, now they are not. Something has changed, and I don't know what.

Switching to older proton version helps to start the game, but the textures are flickering there, so basically unplayable.




## Reproduction
Start game.
Try launching the game.
It freezes.

**Comment by H34dl3ss** [comment]
Hi, we used to play CnC TWKW 1.09 but recently, as some clients switched to Linux, we are running into this out of sync issue when Windows and Linux are in a match.
I understood the procedure above, but we are running Lutris 0.5.16 and all those protontricks things are not applicable.
Otherwise, the game is running fine with wine-ge-8-26.
So I downloaded all the vcruntimes mentioned above and installed them by hand.
Unfortunately it didn't solve the "out of sync" problem.
Any other ideas?

**Comment by Tribalwarfare** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/477#issuecomment-2571257036

How did you install the VC Runtimes? Did you target the correct profile?

**Comment by Tribalwarfare** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/477#issuecomment-2571257036

OK, You can run Winetricks from withing Lutris on the profile :)

See below for instructions: https://www.reddit.com/r/linux_gaming/comments/zw0wft/how_to_install_microsoft_visual_c_runtime_the/

**Comment by deed02392** [comment]
here's an excellent detailed guide to working through what's needed to address desync issues with KW running within Wine/Proton `<Link removed by moderator>`

**Comment by kisak-valve** [comment]
Hello @deed02392, the workaround you linked is legally problematic and has been removed.

**Comment by deed02392** [comment]
thanks @kisak-valve – can you help clarify which part is legally problematic? it still requires the paid-for Steam version, covering tweaks to the Proton compat layer to get things running. everything else is just making the game playable online since GameSpy was shut down.

**Comment by Wedge009** [comment]
I noticed recently - I think with switch to Proton 10 - that the Alt-Tab -> black screen issue seems to be gone.

However, I only just noticed something else (I don't normally play across different computers) - there are two Kane's Wrath directories created, one with an apostrophe and one without:

`<Steam installation>/steamapps/compatdata/24810/pfx/drive_c/users/steamuser/AppData/Roaming/`:
* `Command & Conquer 3 Kane's Wrath`
* `Command & Conquer 3 Kanes Wrath`

The first one, with the apostrophe, is what the game actually uses. The second one, without the apostrophe, is what Steam uses for synchronisation (it contains only `steam_autocloud.vdf`). I'm guessing this is only an issue with Proton (perhaps some path 'cleaning' routine), otherwise users on Windows would have complained long ago that file synchronisation was broken.

I'm checking to see if this is an issue with [Tiberium Wars (24790)](https://github.com/ValveSoftware/Proton/issues/1187) but perhaps not as there is no apostrophe in the title.

**Comment by deed02392** [comment]
Interesting, that sounds like a potential bug, although I suppose if that routine is consistent then it might not cause any functional issues. I need to test this game again on my M1 MacBook Pro, now that Proton 10 is out.

**Comment by kisak-valve** [comment]
Command post not working possibly due to dotnet requirement and 32bit libraries.

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9547.
@ashgaming322-lang posted on 2026-03-06T02:37:07:

Hello! So I'm trying to run CNC3: Kanes wrath through a launcher known as command post. Command post requires dotnet48 & 40. It gives me a warning that wine does not support dotnet outside of 32bit prefixes. So when I try launching command post using "WINEPREFIX="path/to/prefix" "path/to/wine" "path/to/commandpost.exe" ". Command post launches but I get an error saying rundll32.exe could not launch application. Now I can kind of use command post but the start game button fails to work. I was thinking maybe a 32bit prefix would fix this but I dont know how to do that.

Proton ver & Game store:
i have tried proton 9-0-4 as well as the latest glorious eggroll version. Also I have the steam version of the game. 

Additional info:
1. When command post is on, I cant launch the game normally by running it through wine or steam but I can launch it by going into browse folder and double clicking the game.
2. If I launch the game on steam first then launch command post then my game does not crash but again the start button changes to an exit game button and it still does not work.

I know this is not a proton specific issue but I would be grateful if you can help me solve this issue.
***
@ashgaming322-lang commented on 2026-03-06T02:40:22:

<img width="532" height="206" alt="Image" src="https://github.com/user-attachments/assets/c603972a-eb02-4711-990b-4656467bad73" />

**Comment by Wedge009** [comment]
There's a **regression in Proton 11** where custom scenarios and missions downloaded from the workshop are no longer found. Reverting to Proton 10 resolves this.

The same applies for #1187.
